timesketch/api/v1/resources/sketch.py

```diff
@@ -65,28 +65,86 @@ def get(self):
         Returns:
             List of sketches (instance of flask.wrappers.Response)
         """
+        args = self.parser.parse_args()
+        scope = args.get('scope', None)
+        page = args.get('page', 1)
+        search_query = args.get('search_query', None)
+
         if current_user.admin:
             sketch_query = Sketch.query
         else:
             sketch_query = Sketch.all_with_acl()
 
-        filtered_sketches = sketch_query.filter(
+        base_filter = sketch_query.filter(
+            not_(Sketch.Status.status == 'deleted'),
+            not_(Sketch.Status.status == 'archived'),
+            Sketch.Status.parent).order_by(Sketch.updated_at.desc())
+
+        base_filter_with_archived = sketch_query.filter(
             not_(Sketch.Status.status == 'deleted'),
-            Sketch.Status.parent).order_by(Sketch.updated_at.desc()).all()
+            Sketch.Status.parent).order_by(Sketch.updated_at.desc())
 
-        # Just return a subset of the sketch objects to reduce the amount of
-        # data returned.
+        filtered_sketches = base_filter_with_archived
         sketches = []
-        for sketch in filtered_sketches:
-            sketches.append({
+        return_sketches = []
+        num_hits = 0
+
+        if scope == 'recent':
+            views = View.query.filter_by(
+                user=current_user, name='').order_by(
+                View.updated_at.desc()).limit(3)
```

Comment:
why limit with 3?

Where is this magical number derived from? And why is that not configurable?

Reply:
This is a smaller UI window at the top, only with the recent sketches (sketces that you activly queried in). There are other paginated windows below in the UI. It can be made configurable, and I considered that but I don't want to add that now until we have tested the feature a bit

Reply:
can you then at least add a comment or a TODO here to add an option to configure this number.

Reply:
done!